frontend/src/app/event/event-page/event-page.component.ts

```diff
@@ -3,268 +3,156 @@
+    params.filter = this.filterQuery();
+    // Refresh the data
+    this.eventService.getEvents(params).subscribe((events) => {
+      this.page.set(events);
+      this.previousParams = events.params;
+      this.reloadQueryParams();
     });
+  });
+
+  /** Reloads the page and its query parameters to adjust to the next month. */
+  nextPage() {
+    this.startDate.set(
+      new Date(this.startDate().setMonth(this.startDate().getMonth() + 1))
+    );
+    this.endDate.set(
+      new Date(this.endDate().setMonth(this.endDate().getMonth() + 1))
+    );
   }
 
-  /** Handler that runs when an event card is clicked.
-   * This function selects the event to display on the sidebar.
-   * @param event: Event pressed
-   */
-  onEventCardClicked(event: Event) {
-    this.selectedEvent = event;
+  /** Reloads the page and its query parameters to adjust to the previous month. */
+  previousPage() {
+    this.startDate.set(
+      new Date(this.startDate().setMonth(this.startDate().getMonth() - 1))
+    );
+    this.endDate.set(
+      new Date(this.endDate().setMonth(this.endDate().getMonth() - 1))
+    );
   }
 
-  showEvents(isPrevious: boolean) {
-    //let paginationParams = this.page.params;
-    this.startDate = isPrevious
-      ? new Date(this.startDate.setMonth(this.startDate.getMonth() - 1))
-      : new Date(this.startDate.setMonth(this.startDate.getMonth() + 1));
-    this.endDate = isPrevious
-      ? new Date(this.endDate.setMonth(this.endDate.getMonth() - 1))
-      : new Date(this.endDate.setMonth(this.endDate.getMonth() + 1));
-    if (isPrevious === true) {
-      this.page.params.ascending = 'false';
-    }
-    this.today =
-      this.startDate.setHours(0, 0, 0, 0) == new Date().setHours(0, 0, 0, 0);
+  /**
+   * Reloads the page to update the query parameters and reload the data.
+   * This is required so that the correct query parameters are reflected in the
+   * browser's URL field.
+   * @param startDate: The new start date
+   * @param endDate: The new end date
+   */
+  reloadQueryParams() {
     this.router.navigate([], {
       relativeTo: this.route,
       queryParams: {
-        start_date: this.startDate.toISOString(),
-        end_date: this.endDate.toISOString()
+        start_date: this.startDate().toISOString(),
+        end_date: this.endDate().toISOString()
       },
       queryParamsHandling: 'merge'
     });
   }
+
+  // TODO: Refactor this method to remove manual +/- 100 year range on query filtering.
+
+  /** Handler that runs when the search bar query changes.
+   * @param query: Search bar query to filter the items
+   */
+  onSearchBarQueryChange(query: string) {
+    this.startDate.set(
+      query === ''
+        ? new Date()
+        : new Date(new Date().setMonth(new Date().getMonth() + 1))
+    );
+    this.endDate.set(
+      query === ''
+        ? new Date()
+        : new Date(new Date().setMonth(new Date().getMonth() + 1))
+    );
```

Comment:
Slightly confused by what this logic is doing, perhaps a comment would help.

The ternary operator's results seem strange. If query is empty, set start and end to the current date time. Oterwise, if non-empty, set both to one month from now?

Reply:
I agree, I think it would be better to use a regular if-statement here! This ended up like this due to the original code and also a few things being moved around in the refractor.

Reply:
Yeah @KrisJordan, the logic here seems broken - I think it was as follows (and how I just edited it to be):
* If the query is blank, set back to a range of today through today + 1 month
* If the query is not blank, set the range to be (-100 years, +100 years)

I don't necessarily think that this is good, but it is what we had before! I am thinking this rate range should be handled in the backend when a filter query is provided. Thoughts?

Reply:
Got it, makes sense! Ultimately you could imagine wanting these two dimensions to be orthogonal and composable, not either-or (search for this term within this date range) but that becomes a UI problem (and it sounds like this is actually possible via the API logic described).